File: labs OOP/2.2/C#/Lowercase.cs
using System;

namespace StringPolymorphism
{
    public class Lowercase : Strings
    {
        private string value;

        public Lowercase(string s)
        {
            value = s;
        }

        public override int Length()
        {
            return value.Length;
        }

        public override int Offset(char c)
        {
            int pos = value.IndexOf(c);
            return pos;
        }

        public override string GetValue()
        {
            return value;
        }

        public override void Shift()
        {
            if (!string.IsNullOrEmpty(value))
            {
                value = value.Substring(1) + value[0];
            }
        }
    }
}

File: labs OOP/2.2/C++/Lowercase.cpp
#include "Lowercase.h"

Lowercase::Lowercase(const std::string& s) : value(s) {}

int Lowercase::length() const {
    return static_cast<int>(value.length());
}

int Lowercase::position(char c) const {
    size_t pos = value.find(c);
    return (pos != std::string::npos) ? static_cast<int>(pos) : -1;
}

std::string Lowercase::getValue() const {
    return value;
}

void Lowercase::shift() {
    if (!value.empty()) {
        value = value.substr(1) + value.substr(0, 1);
    }
}